### PR TITLE
Generate resourcebinding from clusterpropagationpolicy based on the "namespace" exist of the coming resource

### LINKED ISSUE #300: [HelpWanted] ClusterResourceBinding should only reference cluster-scoped resources
**What would you like to be added**:
When [applying cluster policy](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/util/detector/detector.go#L338) for a resource template, we should build [ResourceBinding](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/apis/work/v1alpha1/binding_types.go#L13) for namespace-scoped resources, and build [ClusterResourceBinding](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/apis/work/v1alpha1/binding_types.go#L102) for cluster-scoped resources.

But currently, we always build [ClusterResourceBinding](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/apis/work/v1alpha1/binding_types.go#L102) and regardless of the resource scope.  The relevant code at :
https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/util/detector/detector.go#L346-L354

**Why is this needed**:
[ResourceBinding](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/apis/work/v1alpha1/binding_types.go#L13) is designed for binding namespace scoped resources and [ClusterResourceBinding](https://github.com/karmada-io/karmada/blob/46de97140ea05de5809f00e337277c209f3b1c2c/pkg/apis/work/v1alpha1/binding_types.go#L102) is designed for binding cluster scoped resources.

### PR DESCRIPTION
**Pull request by vincent-pli**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Since `ClusterPropagationPolicy` is allowed to point to a namespace scoped resource, so in the scenario a `ResourceBinding` rather than `ClusterResourceBinding` should be created.

**Which issue(s) this PR fixes**:
Fixes #300 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

File: pkg/util/detector/detector.go [hunk 1]
```diff
@@ -309,7 +309,12 @@ func (d *ResourceDetector) ApplyPolicy(object *unstructured.Unstructured, object
 		return err
 	}
 
-	binding := d.BuildResourceBinding(object, objectKey, policy)
+	labels := map[string]string{
```
Comment (RainbowMango) [inline pkg/util/detector/detector.go:312]: Variable 'labels' collides with the imported package name.  
How about change it to `policyLabels` or something else?

Same as below in `ApplyClusterPolicy`.

File: pkg/util/detector/detector.go [hunk 2]
```diff
@@ -343,26 +348,58 @@ func (d *ResourceDetector) ApplyClusterPolicy(object *unstructured.Unstructured,
...
-		return err
+	labels := map[string]string{
+		util.ClusterPropagationPolicyLabel: policy.GetName(),
 	}
 
-	if operationResult == controllerutil.OperationResultCreated {
-		klog.Infof("Create ClusterResourceBinding(%s) successfully.", binding.GetName())
-	} else if operationResult == controllerutil.OperationResultUpdated {
-		klog.Infof("Update ClusterResourceBinding(%s) successfully.", binding.GetName())
+	// Namespace scoped resoource will lead to a ResourceBinding; cluster scoped resource will lead to a ClusterResourceBinding
```
Comment (RainbowMango) [inline pkg/util/detector/detector.go:355]: How about giving it more detailed comments, like:
```
	// Build `ResourceBinding` or `ClusterResourceBinding` according to the resource template's scope.
	// For namespace-scoped resources, which namespace is not empty, building `ResourceBinding`.
	// For cluster-scoped resources, which namespace is empty, building `ClusterResourceBinding`.
```

**Review by RainbowMango — COMMENTED** [review]
Just some nits, otherwise LGTM.

There are some duplicate codes in `ApplyClusterPolicy` method, but it's no big deal.

File: pkg/util/detector/detector.go (hunk 1 above)
Comment (vincent-pli) [reply pkg/util/detector/detector.go]: Make sense, thanks @RainbowMango

File: pkg/util/detector/detector.go (hunk 2 above)
Comment (vincent-pli) [reply pkg/util/detector/detector.go]: Done

**Comment by RainbowMango** [comment]
Thanks and congrats! @vincent-pli

**Pull request merged**